src/react-valence-ui-iframe.js

```diff
@@ -17,6 +17,10 @@ var ResizingIframe = React.createClass({
 	},
 	componentDidMount: function() {
 		this.updateProgress(0);
+		this.updateNavbarStyle();
+	},
+	componentDidUpdate: function() {
+		this.updateNavbarStyle();
```

Comment:
If you can check here (or in the updateNavbarStyle) to see if the iframe changed url before running the update, that would prevent this function from running many times on the same page